superset/assets/src/CRUD/CollectionTable.jsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+
+import Button from '../components/Button';
+import Fieldset from './Fieldset';
+import { recurseReactClone } from './utils';
+import './styles.css';
+
+const propTypes = {
+  collection: PropTypes.array,
```

Comment:
can you make any of these more specific? generic `array` and `object`s are not very useful (but it's not always possible to make them more specific either 🤷‍♀️ )

Reply:
also all non-required props should have a default, I don't think that's true here.